compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -1773,8 +1773,6 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
       val returnUnit = lambdaTarget.info.resultType.typeSymbol == defn.UnitClass
       val functionalInterfaceDesc: String = generatedType.descriptor
       val desc = capturedParamsTypes.map(tpe => toTypeKind(tpe)).mkString(("("), "", ")") + functionalInterfaceDesc
-      // TODO specialization
```

Comment:
I think the devil is in this "TODO specialization". I think we should rather go for both signatures specialized rather than unspecialized, what do you think?